docs/sources/flow/reference/components/discovery.http.md

```diff
@@ -27,6 +27,26 @@ Example response body:
 ]
 ```
 
+Example response setting the metrics path and port for [prometheus.scrape](../prometheus.scrape#technical-details) additional config:
+
+```json
+[
+   {
+      "labels" : {
+         "__metrics_path__" : "/api/prometheus",
+         "__scheme__" : "https",
+         "__scrape_interval__" : "60s",
+         "__scrape_timeout__" : "10s",
```

Comment:
I'm leaving this to @clayton-cornell for feedback, but since we're doing this, can we also a) include a `"__param_<name>": "val"` label which will populate a `?name=val` query parameter and b) briefly explain what these parameters do in the above sentence?

Reply:
Happy to tidy up the sentence, didn't realise that you could pass more things to `prometheus.scrape` via labels - is there a full list anywhere?

Main reason for asking is that quite often if `metricsPath` isn't the default, the chances are we'll want to pass a Bearer token or similar as part of the request, but it's not obvious from the docs how to do that on a "per target" basis!

Reply:
Yes; the canonical reference is the upstream Prometheus docs which mention all the special "internal" labels that can be used this way.

https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config

Reply:
> quite often if metricsPath isn't the default, the chances are we'll want to pass a Bearer token or similar as part of the request, but it's not obvious from the docs how to do that on a "per target" basis!

I think that's not available today. The HTTP client configuration is done on a per-component basis, so you'd need to set up multiple `prometheus.scrape` component to authenticate differently for different targets.

Reply:
>I think that's not available today. The HTTP client configuration is done on a per-component basis, so you'd need to set up multiple prometheus.scrape component to authenticate differently for different targets.

Shame - I may file an enhancement for that shortly as well then! 😄 

I've written a small microservice in Django that I'm running with this on my local network, and I keep all the services that aren't in K8s listed in that service:

<img width="1775" alt="image" src="https://github.com/grafana/agent/assets/262545/ed26499c-7f11-4c2d-a0b3-f7daf638bbc9">

The output of the service is a JSON array of hosts, metricsPaths, ports, etc. that can be consumed by `discovery.http` but as soon as any of those endpoints need an auth token, I have to add a new scraper with the token.

The output is as follows at present:

```json
[
   {
      "labels" : {
         "__metrics_path__" : "/metrics",
         "__scheme__" : "http",
         "__scrape_interval__" : "60s",
         "__scrape_timeout__" : "10s",
         "service" : "PiHole"
      },
      "targets" : [
         "192.168.12.1:9100"
      ]
   },
   {
      "labels" : {
         "__metrics_path__" : "/api/prometheus",
         "__scheme__" : "https",
         "__scrape_interval__" : "60s",
         "__scrape_timeout__" : "10s",
         "service" : "HomeAssistant"
      },
      "targets" : [
         "192.168.12.10:443"
      ]
   }
]
```

And this works well for `discovery.http` and a generic `prometheus.scrape` apart from the auth tokens.

Ideally, I'd be able to do something like this:

```json
[
   {
      "labels" : {
         "__metrics_path__" : "/metrics",
         "__scheme__" : "http",
         "__scrape_interval__" : "60s",
         "__scrape_timeout__" : "10s",
         "service" : "PiHole",
         "bearer_token": <MY TOKEN>
      },
      "targets" : [
         "192.168.12.1:9100"
      ]
   },
   {
      "labels" : {
         "__metrics_path__" : "/api/prometheus",
         "__scheme__" : "https",
         "__scrape_interval__" : "60s",
         "__scrape_timeout__" : "10s",
         "service" : "HomeAssistant"
      },
      "targets" : [
         "192.168.12.10:443"
      ]
   }
]
```

and for those targets that support it, prom.scrape would pick that up and use it.

Anyway, I'll log a feature request for that in a bit, but that's the context of how I'm using `discovery.http` 😄

Reply:
Yes, let's open a separate issue for this!

In the meantime, let's also document `__param_<name>` and use the above sentence to explain how the final target will look/behave like to merge this 🙌

Reply:
@tpaschalis - latest push contains additional content, let me know what needs changing?

Reply:
I hadn't seen there was an additional example below, thanks.